### PR TITLE
Hauki 219 sign out

### PR DESCRIPTION
**Pull request by minevala**
- Logout feature
Näyttökuva 2020-11-27 kello 9 53 03

### CODE REVIEW

File: src/components/navigation/HaukiNavigation.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import React, { useState } from 'react';
-import { Navigation } from 'hds-react';
+import { useHistory } from 'react-router-dom';
+import { Navigation, Notification } from 'hds-react';
+import api from '../../common/utils/api/api';
 import { AuthContextProps, useAuth } from '../../auth/auth-context';
 import './HaukiNavigation.scss';
 
 export default function HaukiNavigation(): JSX.Element {
+  const [signOutError, setSignError] = useState<string | undefined>();
```
Comment (mikkokeskinen) [inline src/components/navigation/HaukiNavigation.tsx]: Isn't it customary to use the same name in the variable and the set? i.e. signOutError, setSignOutError.

Comment (mikkokeskinen) [reply src/components/navigation/HaukiNavigation.tsx]: Oh. You changed it already. :D

Comment (minevala) [reply src/components/navigation/HaukiNavigation.tsx]: Yes, I noticed it too 😁

File: src/common/utils/api/api.ts [hunk 2]
```diff
@@ -120,11 +126,24 @@ interface AuthTestResponse {
   username: string;
 }
 
+export interface SuccessResponse {
```
Comment (VilleMiekkoja) [inline src/common/utils/api/api.ts:129]: The name kind of indicates that the response will be always success? But I guess it can be also false?

Comment (minevala) [reply src/common/utils/api/api.ts:129]: Yes, maybe PermissionResponse?

Comment (VilleMiekkoja) [reply src/common/utils/api/api.ts]: That sounds better imo

Comment (minevala) [reply src/common/utils/api/api.ts]: Rename'd

**Review by VilleMiekkoja — APPROVED** [review]

**Pull request merged**